Signal/src/network/PushManager.m

```diff
@@ -44,44 +43,29 @@ + (instancetype)sharedManager {
 
 - (instancetype)initDefault
 {
-    return [self initWithContactsManager:[Environment getCurrent].contactsManager
-                          networkManager:[Environment getCurrent].networkManager
-                          storageManager:[TSStorageManager sharedManager]
-                      callMessageHandler:[Environment getCurrent].callMessageHandler
-                             callService:[Environment getCurrent].callService
-                         contactsUpdater:[Environment getCurrent].contactsUpdater];
+    return [self initWithNetworkManager:[Environment getCurrent].networkManager
+                         storageManager:[TSStorageManager sharedManager]
+                          callUIAdapter:[Environment getCurrent].callService.callUIAdapter
+                        messagesManager:[TSMessagesManager sharedManager]
+                          messageSender:[Environment getCurrent].messageSender];
 }
 
-- (instancetype)initWithContactsManager:(OWSContactsManager *)contactsManager
-                         networkManager:(TSNetworkManager *)networkManager
-                         storageManager:(TSStorageManager *)storageManager
-                     callMessageHandler:(OWSWebRTCCallMessageHandler *)callMessageHandler
-                            callService:(CallService *)callService
-                        contactsUpdater:(ContactsUpdater *)contactsUpdater
+- (instancetype)initWithNetworkManager:(TSNetworkManager *)networkManager
+                        storageManager:(TSStorageManager *)storageManager
+                         callUIAdapter:(CallUIAdapter *)callUIAdapter
+                       messagesManager:(TSMessagesManager *)messagesManager
+                         messageSender:(OWSMessageSender *)messageSender
 {
     self = [super init];
     if (!self) {
         return self;
     }
 
-    _contactsManager = contactsManager;
-    _callUIAdapter = callService.callUIAdapter;
-
-    _messageSender = [[OWSMessageSender alloc] initWithNetworkManager:networkManager
```

Comment:
Much of this complexity goes back to a circular dependency that used to exist between messageSender and messagesManager.

That circular dependency was removed, so this code can be greatly simplified.